pkg/cmd/plugin_package.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright The Helm Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/term"
+
+	"helm.sh/helm/v4/internal/plugin"
+	"helm.sh/helm/v4/pkg/cmd/require"
+	"helm.sh/helm/v4/pkg/provenance"
+)
+
+const pluginPackageDesc = `
+This command packages a Helm plugin directory into a tarball.
+
+By default, the command will generate a provenance file signed with a PGP key.
+This ensures the plugin can be verified after installation.
+
+Use --sign=false to skip signing (not recommended for distribution).
+`
+
+type pluginPackageOptions struct {
+	sign           bool
+	keyring        string
+	key            string
+	passphraseFile string
+	pluginPath     string
+	destination    string
+}
+
+func newPluginPackageCmd(out io.Writer) *cobra.Command {
+	o := &pluginPackageOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "package [PATH]",
+		Short: "package a plugin directory into a plugin archive",
+		Long:  pluginPackageDesc,
+		Args:  require.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			o.pluginPath = args[0]
+			return o.run(out)
+		},
+	}
+
+	f := cmd.Flags()
+	f.BoolVar(&o.sign, "sign", true, "use a PGP private key to sign this plugin")
+	f.StringVar(&o.key, "key", "", "name of the key to use when signing. Used if --sign is true")
+	f.StringVar(&o.keyring, "keyring", defaultKeyring(), "location of a public keyring")
+	f.StringVar(&o.passphraseFile, "passphrase-file", "", "location of a file which contains the passphrase for the signing key. Use \"-\" to read from stdin.")
+	f.StringVarP(&o.destination, "destination", "d", ".", "location to write the plugin tarball.")
```

Comment:
I think the default being `.` is problematic. on my system, when I run
```
/home/jesse/code/helm/bin/helm plugin package --key "Helm Signing Key" --keyring ./exported-keyring.gpg  ./
```

It saves the plugin package to `..tgz.prov` and `..tgz`

```
Successfully signed. Signature written to: ..tgz.prov
Successfully packaged plugin and saved it to: ..tgz
```

makes it hard to find the package.